### ISSUE TITLE
Delegate ReplicateResult

### ISSUE THREAD

**Issue by FintanH** [radicle-dev/radicle-link#499]
When using `replicate` we get back a `ReplicateResult`, currently this has to variants `Latest` and `Behind`. There is no case for when the delegate is `Ahead` and doesn't necessarily need to update -- but maybe it could open up avenues of gossiping to the delegate who is `Behind`.

This also poses the question if we should surface this `ReplicateResult` in the `Gossip` notifications so we can make being `Behind` actionable, i.e. update to the latest.

**Comment by kim** [comment]
Can you explain why this should be broadcasted? And how a receiver should
interpret behind / ahead (relative to...?)?

**Comment by FintanH** [comment]
So just to contextualise where this thought came from when I added `ReplicateResult` it bubbled up to this point https://github.com/radicle-dev/radicle-link/commit/087a1bf48f5b5802878c42986b874fff7cf9b4c0#diff-b650ce3d62e812ad5579e43d0805c510f7324e38f24762a9a81f5466b760993bR546-R548. 
This prompted me to think about whether being `Behind` could be surfaced to the local peer so they can do the update & sign dance. So I'm not thinking about broadcasting in this case.
In the, currently fictitious, case of being `Ahead`, I thought it might be interesting to tell that delegate that you're ahead, somehow. But maybe that's not a use case we want to cover since that delegate will eventually pick up local changes -- since they're both delegates -- through a syncronising fetch or just the gossip layer doing its thing.

**Comment by kim** [comment]
Ya that's a different thing -- we'd need to funnel the result through the ~~enterprise service bus~~channels to allow the library user to react to it. Gossiping doesn't really make sense, because it is converging the state anyways, and what this type describes requires user interaction.

**Comment by FintanH** [comment]
Makes sense :+1: So I would say the scope of this issue is adding the check for determining if the local is `Ahead` instead then.
